react-client/src/routes/CreateTutorial.js

```diff
@@ -29,50 +34,129 @@ export default class CreateTutorial extends Component {
     detailTextCounter: null
   }
 
-  createTut() {
+  createTut = () => {
     this.setState({
-      createTutorial: true
+      createTutorial: !this.state.createTutorial
     })
   }
 
-  createSection() {
+  createSection = (e, index, type) => {
     const tutorialData = { ...this.state.tutorialData }
-    tutorialData.sections.push(sectionsDataModel)
-    this.setState({
-      tutorialData
-    })
+    switch (type) {
+      case "create":
+        tutorialData.sections.push(sectionsDataModel)
+        this.setState({
+          tutorialData
+        })
+        break
+
+      case "delete":
+        tutorialData.sections.splice(index, 1)
+        this.setState({
+          tutorialData
+        })
+        break
+    }
+  }
+
+  SectionDetailsHandler = (e, index, type) => {
```

Comment:
if we're gonna put type and index instead of i and t, how come e isn't event? ;) :P

Reply:
Fixed it.